packages/block-library/src/block/edit.js

```diff
@@ -47,14 +57,37 @@ export default function ReusableBlockEdit( { attributes: { ref }, clientId } ) {
 				'wp_block',
 				ref,
 			] );
+			const blockHasEdits = select( coreStore ).hasEditsForEntityRecord(
+				'postType',
+				'wp_block',
+				ref
+			);
 			return {
 				hasResolved: hasResolvedBlock,
 				isMissing: hasResolvedBlock && ! persistedBlock,
+				hasEdits: blockHasEdits,
 			};
 		},
 		[ ref, clientId ]
 	);
 
+	const { isChildSelected, isSelected } = useSelect( ( select ) => {
+		const { getBlockParents, getSelectedBlockClientId } = select(
+			blockEditorStore
+		);
+		const selectedBlockClientId = getSelectedBlockClientId();
+		const _isSelected = selectedBlockClientId === clientId;
+
+		const parents = getBlockParents( selectedBlockClientId );
+		const firstParentClientId = parents[ parents.length - 1 ];
+		const _isChildSelected = firstParentClientId === clientId;
```

Comment:
Shouldn't the case of changing nested blocks inside the `reusable` block been handled somehow? I was expecting here to see a `getBlockRootClientId` and check if is `reusable` and `hasEdits`... If we add a `Group` with a `Paragraph` inside the reusable block and change that paragraph, the block has edits but there is no indication for the changes. Of course in this case the `Group` block is the parent.. I'm not sure how it would be best to show this.

Reply:
> I was expecting here to see a getBlockRootClientId and check if is reusable and hasEdits

Yes, this is much cleaner and better. Thank you for the suggestion. Updating this in my next commit.

> If we add a Group with a Paragraph inside the reusable block and change that paragraph, the block has edits but there is no indication for the changes. Of course in this case the Group block is the parent.. I'm not sure how it would be best to show this.

Even I am not sure about this case cause one has to select the Reusable block itself to access the save button in the block toolbar.

Reply:
>If we add a Group with a Paragraph inside the reusable block and change that paragraph, the block has edits but there is no indication for the changes. 

Introducing dot indicators into List view has come up as one possible solution that would help with more visibility in instances with a lot of nesting. Needs a design explore but I could look at this alongside other follow ups in https://github.com/WordPress/gutenberg/issues/33988!